lib/sequent/core/event_store.rb

```diff
@@ -174,110 +161,104 @@ def replay_events_from_cursor(get_events:, block_size: 2000,
         raise DeserializeEventError, event_hash
       end
 
-      def resolve_event_type(event_type)
-        event_types.fetch_or_store(event_type) { |k| Class.const_get(k) }
-      end
-
       def publish_events(events)
         Sequent.configuration.event_publisher.publish_events(events)
       end
 
       def store_events(command, streams_with_events = [])
-        command_record = CommandRecord.create!(command: command)
-        event_records = streams_with_events.flat_map do |event_stream, uncommitted_events|
-          unless event_stream.stream_record_id
-            stream_record = Sequent.configuration.stream_record_class.new
-            stream_record.event_stream = event_stream
-            stream_record.save!
-            event_stream.stream_record_id = stream_record.id
-          end
-          uncommitted_events.map do |event|
-            Sequent.configuration.event_record_class.new.tap do |record|
-              record.command_record_id = command_record.id
-              record.stream_record_id = event_stream.stream_record_id
-              record.event = event
-            end
-          end
+        command_record = {
+          created_at: convert_timestamp(command.created_at&.to_time || Time.now),
+          command_type: command.class.name,
+          command_json: command,
+        }
+
+        events = streams_with_events.map do |stream, uncommitted_events|
+          [
+            Sequent::Core::Oj.strict_load(Sequent::Core::Oj.dump(stream)),
+            uncommitted_events.map do |event|
+              {
+                created_at: convert_timestamp(event.created_at.to_time),
+                event_type: event.class.name,
+                event_json: event,
+              }
+            end,
+          ]
         end
-        connection = Sequent.configuration.event_record_class.connection
-        values = event_records
-          .map { |r| "(#{column_names.map { |c| connection.quote(r[c.to_sym]) }.join(',')})" }
-          .join(',')
-        columns = column_names.map { |c| connection.quote_column_name(c) }.join(',')
-        sql = <<~SQL.chomp
-          insert into #{connection.quote_table_name(Sequent.configuration.event_record_class.table_name)} (#{columns}) values #{values}
-        SQL
-        Sequent.configuration.event_record_class.connection.insert(sql, nil, primary_key_event_records)
+        call_procedure(
+          'store_events',
+          [
+            Sequent::Core::Oj.dump(command_record),
+            Sequent::Core::Oj.dump(events),
+          ],
+        )
       rescue ActiveRecord::RecordNotUnique
         raise OptimisticLockingError
       end
+
+      def convert_timestamp(timestamp)
+        # Since ActiveRecord uses `TIMESTAMP WITHOUT TIME ZONE`
+        # we need to manually convert database timestamps to the
+        # ActiveRecord default time zone on serialization.
+        ActiveRecord.default_timezone == :utc ? timestamp.getutc : timestamp.getlocal
```

Comment:
Perhaps we should document this that we fallback to the `:local` timezone in case it is not `utc`? Or shouldn't we use the one set in `default_timezone`?

Reply:
The ` ActiveRecord.default_timezone` can only have the values `:local` or `:utc`, AFAIK. Here is some description of how Rails/ActiveRecord is configured for time zones: https://stackoverflow.com/a/32229086